src/videojs.ads.js

```diff
@@ -147,33 +143,19 @@ var
           player.currentTime(snapshot.ended ? player.duration() : snapshot.currentTime);
         }
 
+        // Android reports duration Infinity right after switching source, so we need
+        // this to determine if ad was a postroll.
+        var androidPostroll = snapshot.ended &&
```

Comment:
Can we add more info to the snapshot to identify this? This condition seems racy to me.